airflow/datasets/__init__.py

```diff
@@ -336,6 +342,14 @@ def iter_datasets(self) -> Iterator[tuple[str, Dataset]]:
                 yield k, v
                 seen.add(k)
 
+    def iter_dataset_aliases(self) -> Iterator[DatasetAlias]:
+        """Filter dataest aliases in the condition."""
+        for o in self.objects:
+            if isinstance(o, _DatasetAliasCondition):
+                yield DatasetAlias(o.name)
```

Comment:
Since `_DatasetAliasCondition` already implements this yield, the if-else condition does not seem needed? This can just be

```python
for o in self.objects:
    yield from o.iter_dataset_aliases()
```

Reply:
Indeed, miss this one. Thanks for reminding me. Just removed it!